cow_py/contracts/sign.py

```diff
@@ -0,0 +1,104 @@
+from enum import IntEnum
+from typing import List, NamedTuple, Union
+
+from eth_account import Account
+from eth_account.datastructures import SignedMessage
+from eth_account.signers.local import LocalAccount
+from eth_utils.conversions import to_hex
+from eth_utils.crypto import keccak
+from web3 import Web3
+
+from cow_py.contracts.order import (
+    CANCELLATIONS_TYPE_FIELDS,
+    ORDER_TYPE_FIELDS,
+    Order,
+    hash_typed_data,
+    normalize_order,
+)
+
+EIP1271_MAGICVALUE = to_hex(keccak(text="isValidSignature(bytes32,bytes)"))[:10]
+PRE_SIGNED = to_hex(keccak(text="GPv2Signing.Scheme.PreSign"))
+
+
+class SigningScheme(IntEnum):
+    # The EIP-712 typed data signing scheme. This is the preferred scheme as it
+    # provides more infomation to wallets performing the signature on the data
+    # being signed.
+    #
+    # https://github.com/ethereum/EIPs/blob/master/EIPS/eip-712.md#definition-of-domainseparator
+    EIP712 = 0b00
+    # Message signed using eth_sign RPC call.
+    ETHSIGN = 0b01
+    # Smart contract signatures as defined in EIP-1271.
+    EIP1271 = 0b10
+    # Pre-signed order.
+    PRESIGN = 0b11
+
+
+class EcdsaSignature(NamedTuple):
+    scheme: SigningScheme
+    data: str
+
```

Comment:
Sorry - why do we use NamedTuples here and in other places Dataclasses?
If I understand correctly, they are both container classes, right?